events/rancher_state_watcher.go

```diff
@@ -0,0 +1,172 @@
+package events
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/fsouza/go-dockerclient"
+	"gopkg.in/fsnotify.v1"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+const healthCheckFileName = ".healthcheck"
+
+var healthCheckTimeout = time.Second * 10
+var healthCheckWriteInterval = time.Second * 8
+var restartWaitUnit = time.Second
+var maxRestarts = 5
+
+func watchRancherState(eventChannel chan<- *docker.APIEvents, watchDir string) {
+	// Ideally, I'd prefer to just bubble an error all the way up  with no restart logic
+	// until it causes the process to exit (and be restared)t, but since this code currently
+	// lives with the rest of host-api, I don't feel comfortable being so cavalier with process exits.
+	// So this will try to restart a number of times, waiting expontentially
+	restarts := 0
+	restartWait := 0
+	log.Infof("Watching state directory: %v", watchDir)
+	for {
+		if restarts >= maxRestarts {
+			panic("Unable to successfully start rancher state watcher.")
+		}
+		if err := watchRancherStateInternal(eventChannel, watchDir); err != nil {
+			log.Warnf("Rancher state watcher returned with error. Waiting %v and then restarting. "+
+				"Error that caused exit: %v", restartWait, err)
+			time.Sleep(restartWaitUnit * time.Duration(restartWait))
+			if restartWait == 0 {
+				restartWait = 2
+			} else {
+				restartWait *= restartWait
```

Comment:
I think you want to multiply by two, so basically 2^X-1 where 0 <= X < max.  This logic will wait, 0, 2, 4, 16, 256.  So the last wait is 4 minutes.